A .gitignore file should be added

So that building VeraCrypt doesn't result in tons of files being reported as "untracked":

```
(master)VeraCrypt $ LC_ALL=C git status 
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

    src/Common/Language.xml.h
    src/Common/Textual_logo_96dpi.bmp.h
    src/Format/VeraCrypt_Wizard.bmp.h
    src/License.txt.h
    src/Main/SystemPrecompiled.d
    src/Main/veracrypt_install_console_x64.sh
    src/Main/veracrypt_install_gui_x64.sh
    src/Mount/Drive_icon_96dpi.bmp.h
    src/Mount/Drive_icon_mask_96dpi.bmp.h
    src/Mount/Logo_96dpi.bmp.h
    src/Platform/Buffer.d
    src/Platform/Buffer.o
    src/Platform/Event.d
    src/Platform/Event.o
[...]
    src/Volume/EncryptionAlgorithm.d
    src/Volume/EncryptionAlgorithm.o
    src/Volume/EncryptionMode.d
    src/Volume/EncryptionMode.o
    src/wxrelease/

nothing added to commit but untracked files present (use "git add" to track)
```

Apparently, VeraCrypt got new dependencies, and I can't even build (Fedora 31):
```
$ make package 
make[1]: wx-config: Command not found
make[1]: wx-config: Command not found
Precompiling SystemPrecompiled.h
SystemPrecompiled.h:13:10: fatal error: wx/wx.h: No such file or directory
   13 | #include <wx/wx.h>
      |          ^~~~~~~~~
compilation terminated.
make[1]: *** [/home/fd/git/VeraCrypt/src/Build/Include/Makefile.inc:50: SystemPrecompiled.h.gch] Error 1
make: *** [Makefile:434: package] Error 2
[fd@linux src]$ uname -a
Linux linux.fritz.box 5.4.13-201.fc31.x86_64 #1 SMP Tue Jan 21 17:21:47 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ make clean
Cleaning Platform
Cleaning Volume
Cleaning Driver
Cleaning Core
make[1]: wx-config: Command not found
Cleaning Main
```

But the .gitignore file is definitely still missing.

VeraCrypt developers should know much better than me which files should be ignored by git. Basically, at least all files that will be created from the source files during the build process, should be ignored.

A good start might be ignoring *.o and *.d files. (You wouldn't want those files ending up in a git repository, would you?)
```
diff --git a/.gitignore b/.gitignore
new file mode 100644
index 00000000..31dc3078
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,2 @@
+*.d
+*.o
```

See `man 5 gitignore` for more details.

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,18 @@
+# For those using Visual Studio Code for development
+.vscode/
+
+# VC Linux build artifacts
+*.o
+*.o0
+*.d
+*.a
+*.bmp.h
+*.xml.h
+*.txt.h
+*.h.gch
+src/Main/veracrypt
+
+# wxWidgets Linux build artifacts
+src/wxrelease
+src/wxdebug
+

